sdk/highlight-node/src/client.ts

```diff
@@ -212,4 +258,36 @@ export class Highlight {
 			.forceFlush()
 			.catch((e) => console.warn('highlight-node failed to flush: ', e))
 	}
+
+	async waitForFlush() {
+		return new Promise<void>(async (resolve) => {
+			let resolved = false
+			let waitingForFinishedSpans = false
+
+			await this.flush()
+
+			let intervalTimer = setInterval(async () => {
+				const finishedSpansCount = this.finishedSpans.length
+
+				if (finishedSpansCount) {
+					waitingForFinishedSpans = true
+				} else if (waitingForFinishedSpans) {
+					finish()
+				}
+			}, 100)
+			let timer = setTimeout(finish, 10000)
```

Comment:
so this effectively forces a finish here if the intervalTimer doesn't have enough spans within 10 seconds?

Reply:
Exactly. I've never seen it take more than ~7000ms to flush everything. It's usually much faster. But if anything goes wrong, the serverless function will stay alive until it's timed-out by Vercel, which is about 60s. It's mostly to backstop my own testing, but it seems reasonable to leave it in.